src/test/unit_tests/test_s3_ops.js

```diff
@@ -94,51 +97,61 @@ mocha.describe('s3_ops', function() {
             if (bucket_type === "regular") {
                 await s3.createBucket({ Bucket: bucket_name }).promise();
                 await s3.createBucket({ Bucket: source_bucket }).promise();
-            } else if (SKIP_TEST) {
-                this.skip(); // eslint-disable-line no-invalid-this
             } else {
+                REMOTE_SOURCE_BUCKET = caching ? SOURCE_BUCKET + '-caching' : SOURCE_BUCKET;
+                REMOTE_TARGET_BUCKET = caching ? TARGET_BUCKET + '-caching' : TARGET_BUCKET;
+
                 const read_resources = [RESOURCE_NAME];
                 const write_resource = RESOURCE_NAME;
                 const read_resources1 = [RESOURCE_NAME_SOURCE];
                 const write_resource1 = RESOURCE_NAME_SOURCE;
-                const ENDPOINT = process.env.ENDPOINT ? process.env.ENDPOINT : 'https://s3.amazonaws.com';
-                const ENDPOINT_TYPE = process.env.ENDPOINT_TYPE ? process.env.ENDPOINT_TYPE : 'AWS';
+                const ENDPOINT = USE_REMOTE_ENDPOINT ? process.env.ENDPOINT : coretest.get_https_address();
+                const ENDPOINT_TYPE = USE_REMOTE_ENDPOINT ? process.env.ENDPOINT_TYPE : 'AWS';
+                const AWS_ACCESS_KEY_ID = USE_REMOTE_ENDPOINT ? process.env.AWS_ACCESS_KEY_ID : s3.config.accessKeyId;
+                const AWS_SECRET_ACCESS_KEY = USE_REMOTE_ENDPOINT ? process.env.AWS_SECRET_ACCESS_KEY : s3.config.secretAccessKey;
+
+                coretest.log("Using endpoint: ", ENDPOINT);
+
                 await rpc_client.account.add_external_connection({
-                        name: CONNECTION_NAME,
-                        endpoint: ENDPOINT,
-                        endpoint_type: ENDPOINT_TYPE,
-                        identity: process.env.AWS_ACCESS_KEY_ID,
-                        secret: process.env.AWS_SECRET_ACCESS_KEY,
+                    name: CONNECTION_NAME,
+                    endpoint: ENDPOINT,
+                    endpoint_type: ENDPOINT_TYPE,
+                    identity: AWS_ACCESS_KEY_ID,
+                    secret: AWS_SECRET_ACCESS_KEY,
                 });
 
+                /**
+                 * NOTE: Remote buckets are not created in the test. It is assumed that
+                 * a. The buckets exist
+                 * b. The buckets are empty
+                 * Following buckets need to exist on the remote endpoint:
+                 *
+                 * remote-s3-ops-bucket-caching
+                 * remote-s3-ops-source-caching
+                 */
+                if (!USE_REMOTE_ENDPOINT) {
```

Comment:
Shouldn't this be the opposite condition to decide if to create the remote buckets?
```suggestion
                if (USE_REMOTE_ENDPOINT) {
```

Reply:
We need 2 buckets for both remote and non remote endpoint.
For `USE_REMOTE_ENDPOINT` the buckets are assumed to exist on the remote endpoint.
For a non remote endpoint, the buckets are created in noobaa.

The remote endpoint's buckets are an issue as there can be multiple pull requests running this test, and all of the pull requests will use the same buckets for reading/writing objects with the same name. This will cause conflicts and fail the tests. The ideal solution will be for the remote buckets have a unique name per pull request test (say the commit id).

This is not an issue when a non remote endpoint is used as the noobaa core gets set up for the specific pull request test so the bucket name has isolation 

In the above code, 2 buckets are created only if `USE_REMOTE_ENDPOINT` is false, then the tests run on noobaa based buckets